test/e2e/gateway_api_test.go

```diff
@@ -0,0 +1,189 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	gwapi "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/gatewayapi"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	testHostname = "*.example.com"
+)
+
+// Define here so we can delete them in the defer section
+var gateway v1beta1.Gateway
+var gatewayClass v1beta1.GatewayClass
+
+// TestGatewayAPIResources tests that basic functions for Gateway API Custom Resources are functional.
+// It specifically verifies that when the GatewayAPI feature gate is enabled, then a user can
+// create a GatewayClass, Gateway, and HTTPRoute.
+func TestGatewayAPIResources(t *testing.T) {
+	t.Parallel()
+
+	// Get the cluster feature gate
+	featureGate := &configv1.FeatureGate{}
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		name := types.NamespacedName{"default", "cluster"}
```

Comment:
The FeatureGate is cluster-scoped, not namespaced.  We usually use the empty string to represent the namespace of cluster-scoped resources.  I'm curious whether specifying the "default" namespace will work here.

Reply:
It did work, but I can change this.

Reply:
done